delivery.go

```diff
@@ -171,3 +190,38 @@ func (d Delivery) Nack(multiple, requeue bool) error {
 	}
 	return d.Acknowledger.Nack(d.DeliveryTag, multiple, requeue)
 }
+
+type DeliveryResponse uint8
+
+const (
+	Ack DeliveryResponse = iota
+	Reject
+	Nack
+)
+
+func (r DeliveryResponse) Name() string {
+	switch r {
+	case Ack:
+		return "ack"
+	case Nack:
+		return "nack"
+	case Reject:
+		return "reject"
+	default:
+		return "unknown"
+	}
+}
+
+func (d *Delivery) Settle(ctx context.Context, response DeliveryResponse, multiple, requeue bool) error {
```

Comment:
I feel deeply ambivalent about this approach, but the alternative would seem to be providing `(Ack,Nack,Rject)Ctx` methods. TBH that's probably better.

Reply:
I agree that `(Ack,Nack,Reject)Ctx` methods are probably a better alternative, and settle the delivery automagically in those functions.